src/utils/reconnection-notifier.js

```diff
@@ -0,0 +1,53 @@
+var C = require( '../constants/constants' );
+
+/**
+ * Makes sure that all functionality is resubscribed on reconnect. Subscription is called
+ * when the connection drops - which seems counterintuitive, but in fact just means
+ * that the re-subscription message will be added to the queue of messages that
+ * need re-sending as soon as the connection is re-established.
+ * 
+ * Resubscribe logic should only occur once per connection loss
+ *
+ * @param {Client} client          The deepstream client
+ * @param {Function} reconnect     Function to call to allow resubscribing
```

Comment:
(Nitpicking) could we call this method (and associate this._reconnect) "resubscribe" please